rocketpool-cli/commands/pdao/status.go

```diff
@@ -0,0 +1,194 @@
+package pdao
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/urfave/cli/v2"
+
+	"github.com/rocket-pool/node-manager-core/eth"
+	utilsMath "github.com/rocket-pool/node-manager-core/utils/math"
+	"github.com/rocket-pool/rocketpool-go/v2/types"
+	"github.com/rocket-pool/rocketpool-go/v2/utils/strings"
+	"github.com/rocket-pool/smartnode/v2/rocketpool-cli/client"
+	"github.com/rocket-pool/smartnode/v2/shared/types/api"
+)
+
+const (
+	colorBlue  string = "\033[36m"
+	colorReset string = "\033[0m"
+	colorGreen string = "\033[32m"
+)
+
+func getStatus(c *cli.Context) error {
+
+	// Get RP client
+	rp, err := client.NewClientFromCtx(c)
+	if err != nil {
+		return err
+	}
+
+	// Get PDAO status at the latest block
+	response, err := rp.Api.PDao.GetStatus()
+	if err != nil {
+		return err
+	}
+
+	// Get Protocol DAO proposals
+	allProposals, err := rp.Api.PDao.Proposals()
+	if err != nil {
+		return err
+	}
+
+	// Get protocol DAO proposals
+	claimableBondsResponse, err := rp.Api.PDao.GetClaimableBonds()
+	if err != nil {
+		fmt.Errorf("error checking for claimable bonds: %w", err)
+	}
+	claimableBonds := claimableBondsResponse.Data.ClaimableBonds
+
+	// Snapshot voting status
+	fmt.Printf("%s=== Snapshot Voting ===%s\n", colorGreen, colorReset)
+	blankAddress := common.Address{}
+	if response.Data.SnapshotVotingDelegate == blankAddress {
+		fmt.Println("The node does not currently have a voting delegate set, which means it can only vote directly on Snapshot proposals (using a hardware wallet with the node mnemonic loaded).\nRun `rocketpool n sv <address>` to vote from a different wallet or have a delegate represent you. (See https://delegates.rocketpool.net for options)")
+	} else {
+		fmt.Printf("The node has a voting delegate of %s%s%s which can represent it when voting on Rocket Pool Snapshot governance proposals.\n", colorBlue, response.Data.SnapshotVotingDelegateFormatted, colorReset)
+	}
+
+	if response.Data.SnapshotResponse.Error != "" {
+		fmt.Printf("Unable to fetch latest voting information from snapshot.org: %s\n", response.Data.SnapshotResponse.Error)
+	} else {
+		voteCount := 0
+		// for _, activeProposal := range response.Data.SnapshotResponse.ActiveSnapshotProposals {
```

Comment:
I'm generally not a fan of leaving commented code blocks in the source.

Either delete, or justify it with a plaintext double-comment explaining why it was commented out, eg:

`// // This block was commented out because it is no longer blah blah but we don't want to delete it yet because it contains the secret to life itself`